install_dependencies.sh

```diff
@@ -36,7 +36,29 @@ echo "Installing Misc. Utilities"
 echo "================================================================"
 
 sudo apt-get install -y\
-    clang-format
+    clang-format\
+    python-rosinstall
+
+
+echo "================================================================"
+echo "Installing Project Dependent ROS packages."
+echo "================================================================"
+
+# Setup rosinstall
+sudo mkdir -p /usr/share/ros/
+sudo chmod a+rwx /usr/share/ros
+rosinstall .
+echo "bash $CURR_DIR/setup.sh" >> ~/.bashrc
```

Comment:
To prevent adding another copy of this line to the `.bashrc` every time we run this script, and to support multiple shells (bash, zsh, etc.), could we add something here like we have in the Snowbots repo `get_started.sh` script? Something like:
```
echo "================================================================"
echo "Setting up your shell config files"
echo "================================================================"
# Shell config files that various shells source when they run.
# This is where we want to add aliases, source ROS environment
# variables, etc.
SHELL_CONFIG_FILES=(
    "$HOME/.bashrc"\
    "$HOME/.zshrc"
)

# All lines listed here will be added to the shell config files
# listed above, if they are not present already
declare -a new_shell_config_lines=(
    "source $CURR_DIR/setup.sh"
)

# Add all of our new shell config options to all the shell
# config files, but only if they don't already have them
for file_name in "${SHELL_CONFIG_FILES[@]}"; 
do
    echo "Setting up $file_name"
    for line in "${new_shell_config_lines[@]}"; 
    do
        if ! grep -Fq "$line" $file_name 
        then
            echo "$line" >> $file_name 
        fi
    done
done
```
Also, I think we should be using `source` instead of `bash`, as normally these sorts of `setup` scripts set useful environment variables in whatever shell they're run in, and I *think* `bash` spawns another shell to run the process in.

Reply:
I copied, assuming everything is correct